index.html

```diff
@@ -0,0 +1,475 @@
+        </ul>
+      </td>
+    </tr>
+  </table>
+</header>
+
+## Introduction: What is CSDL?
+
+```mermaid
+#::[Illustration: CSDL composes multiple subgraph schemas.]
+graph LR
+  s1(astronauts.graphql)-->CSDL
+  s2(rockets.graphql)-->CSDL
+  s3(robots.graphql)-->CSDL
+  style CSDL fill:none,stroke:fuchsia,color:fuchsia;
+```
+
+The Composed Schema Definition Layer (CSDL) provides schema elements to describe a GraphQL Schema which is composed from one or more other GraphQL schemas. **Schema elements** comprise any item in a GraphQL schema—types, input types, directives, etc. CSDL defines directives and mandatory input types. 
+
+The CSDL is intended to be a single artifact which can power a graph router. It declares the composed schema and annotates it with directives that allow the router to disassemble input queries and dispatch them to subgraphs.
+
+CSDL restricts the [GraphQL interface description language](https://spec.graphql.org/) in certain ways: type extensions are not permitted, for instance. It also requires the definition of certain types and provides a suite of directives to describe graph topology. In particular, a CSDL can:
+- define [subgraphs](#def-subgraph) and bind them to [endpoints](#def-endpoint) (with the required [`csdl_Graph`](#csdl_Graph) type and [`@csdl_endpoint`](#csdl_endpoint) directive)
+- assign fields to subgraphs (with [`@csdl_resolve`](#csdl_resolve))
+- declare additional data required and provided by subgraph field resolvers (with [`@csdl_key`](#csdl_key), and [`@csdl_resolve`](#csdl_resolve))
+
+## How to read this document
+
+This document uses [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt) guidance regarding normative terms:
+
+> The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+> NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and
+> "OPTIONAL" in this document are to be interpreted as described in
+> RFC 2119.
+
+### What this document isn't
+
+This document specifies only the structure and semantics of the CSDL. It's expected that the CSDL will be the output of a compilation process which composes subgraph schemas. The mechanicss of that process are not specified normatively here; a suggestion is provided in [Appendix: Basic Composition Algorithm](#appendix-suggested-composition-algorithm). Conforming implementations may choose any approach they like, so long as the result is conforming CSDL.
+
+## Example: Let's go to the moon
+
+*This section is non-normative.*
+
+We'll refer to this example throughout the document. The example consists of two example schemas—a `rockets` schema that holds data about rockets, and an `astronauts` schema that holds information about the people flying on them—along with a CSDL that composes them.
+
+```graphql
+#::[Rockets example service]
+#::a.not-csdl
+type Query {
+  rockets: [Rocket]!
+}
+
+type Rocket @key(fields: "id") {
+  id: String! 
+  name: String!
+  captain: Astronaut @provides(fields: "tripId")
```

Comment:
I'm finding this example a bit confusing: I'm not sure I understand why Rocket has two Astronaut[!] fields, one provided by one service and one by the other, like just at a motivational level.

Reply:
Maybe this:
```
extend type Rocket @key(fields: "id") {
  id: String! @external
  astronaut: Astronaut!
}
```

can just be 
```
extend type Rocket @key(fields: "id") {
  id: String! @external
  crew: [Astronaut]
}
```